torchgeo/datasets/utils.py

```diff
@@ -482,3 +482,31 @@ def rgb_to_mask(
         if isinstance(cmask, np.ndarray):
             mask[cmask.all(axis=-1)] = i
     return mask
+
+
+def percentile_normalization(
+    img: np.ndarray, lower: float = 2, upper: float = 98  # type: ignore[type-arg]
+) -> np.ndarray:  # type: ignore[type-arg]
+    """Applies percentile normalization to an input image.
+
+    Specifically, this will rescale the values in the input such that values <= the
+    lower percentile value will be 0 and values >= the upper percentile value will be 1.
+    Using the 2nd and 98th percentile usually results in good visualizations.
+
+    Args:
+        img: image to normalize
+        lower: lower percentile in range [0,100]
+        upper: upper percentile in range [0,100]
```

Comment:
Should we assert that the values are actually in this range? Does numpy do this for us?

Reply:
numpy throws a value error if they are out of [0,100]